src/pybind/mgr/prometheus/module.py

```diff
@@ -866,6 +867,19 @@ def get_server_addr(self) -> str:
             return self.get_mgr_ip()
         return server_addr
 
+    def wait_for_port_release(self, port, host='127.0.0.1', timeout=60):
+        """Wait until the given port is no longer in use."""
+        end_time = time.time() + timeout
+        while time.time() < end_time:
+            self.log.info('waiting to fully stop...')
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+                try:
+                    s.bind((host, port))
```

Comment:
I will add a log line pointing the socket is not used

Reply:
Done